File: agent-service/app/models/task.py
from sqlalchemy import Column, String, Integer, Date, DateTime, ForeignKey, JSON, Index
from sqlalchemy.sql import func
from app.models.base import Base
import uuid


class Task(Base):
    __tablename__ = "tasks"
    __table_args__ = (Index('idx_user_date', 'user_id', 'task_date'),)

    id = Column(String, primary_key=True, default=lambda: str(uuid.uuid4()))
    plan_id = Column(String, ForeignKey("study_plans.id"), nullable=False)
    user_id = Column(String, ForeignKey("users.id"), nullable=False)
    task_date = Column(Date, nullable=False)
    subject = Column(String, nullable=False)
    task_content = Column(String, nullable=False)
    estimated_minutes = Column(Integer, default=60)
    task_type = Column(Integer, default=1)   # 1-学习 2-复习 3-测试 4-其他
    phase = Column(Integer, default=1)
    status = Column(Integer, default=0)      # 0-待完成 1-进行中 2-已完成
    actual_minutes = Column(Integer, nullable=True)
    completed_at = Column(DateTime, nullable=True)
    keywords = Column(JSON, default=list)    # 用于打卡匹配
    created_at = Column(DateTime, server_default=func.now())

File: backend/src/main/java/com/prepkeeper/entity/Task.java
package com.prepkeeper.entity;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDate;
import java.time.LocalDateTime;

/**
 * 任务实体
 */
@Data
@Entity
@Table(name = "tasks")
public class Task {

    @Id
    @Column(name = "task_id", length = 64)
    private String taskId;

    @Column(name = "plan_id", length = 64, nullable = false)
    private String planId;

    @Column(name = "task_date", nullable = false)
    private LocalDate taskDate;

    @Column(name = "subject", length = 50, nullable = false)
    private String subject;

    @Column(name = "task_content", length = 500, nullable = false)
    private String taskContent;

    @Column(name = "estimated_minutes")
    private Integer estimatedMinutes;

    /**
     * 任务类型: 1-学习 2-复习 3-刷题 4-模考
     */
    @Column(name = "task_type")
    private Integer taskType = 1;

    /**
     * 所属阶段: 1-基础 2-强化 3-冲刺
     */
    @Column(name = "phase", nullable = false)
    private Integer phase;

    /**
     * 状态: 0-未开始 1-进行中 2-已完成 3-已跳过
     */
    @Column(name = "status")
    private Integer status = 0;

    @Column(name = "completed_at")
    private LocalDateTime completedAt;

    /**
     * 打卡方式: 1-文字 2-图片
     */
    @Column(name = "checkin_type")
    private Integer checkinType;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
    }
}